### ISSUE TITLE
The issue of cudnn affecting speed

### ISSUE THREAD

**Issue by iaixuexi** [MzeroMiko/VMamba#204]
Hello, we tried to turn off cudnn, but there was no improvement in speed. Does it need to be in a specific version (such as V3) to be effective?
![image](https://github.com/MzeroMiko/VMamba/assets/51728620/82a60b85-46f3-4eaf-bfd6-b28911ca5585)

**Comment by MzeroMiko** [comment]
Do not worry, because it only happens to some machines (I do not get the pattern actually why a machine will be slow, maybe it is related to the driver or library).

using `torch.backends.cudnn.enabled=True` in downstream tasks may be quite slow. If you found vmamba quite slow in your machine, disable it in vmamba.py, else, ignore this.

**Comment by LQchen1** [comment]
![image](https://github.com/MzeroMiko/VMamba/assets/93528700/4ab2cce7-892b-4178-8094-b032c07377f5)
I had a similar problem, even though I set it to False in vmamba.py, its training time was still unstable, I used pytorch==2.0.1, python=3.11, V100
![image](https://github.com/MzeroMiko/VMamba/assets/93528700/e4791e0b-6802-4b04-abc6-1de8bd10cfac)

**Comment by LQchen1** [comment]
> Do not worry, because it only happens to some machines (I do not get the pattern actually why a machine will be slow, maybe it is related to the driver or library).
> 
> using `torch.backends.cudnn.enabled=True` in downstream tasks may be quite slow. If you found vmamba quite slow in your machine, disable it in vmamba.py, else, ignore this.

In fact, when I used an A100*8 card with batch size set to 512, it took me 2 hours to run an epoch。
![image](https://github.com/MzeroMiko/VMamba/assets/93528700/1ee1f9a7-61e0-4cd5-af3b-519d66ec0bb8)

**Comment by MzeroMiko** [comment]
That seems not possible. What environment you are using?

**Comment by MzeroMiko** [comment]
Also, for 8xV100, the time is about 10mins per epoch.

**Comment by LQchen1** [comment]
> Also, for 8xV100, the time is about 10mins per epoch.

I re-executed the program, and the first round took a long time, but the subsequent training was normal. However, I set torch.backends.cudnn.enabled = True. Could you let me know if this has a big impact on the model's performance?

**Comment by LQchen1** [comment]
> Also, for 8xV100, the time is about 10mins per epoch.

Thank you for your contribution. in addition，In the SS2D model, the author initialized the A matrix to 0. However, in Mamba's paper, I found that the HIPPO initialization of the A matrix is ​​crucial to the performance of the model. I would like to ask why the author did not adopt this method?

**Comment by MzeroMiko** [comment]
It is still weird somehow, and I do not know why the first round would be abnormal. 
In my experiments, all epochs' time-consumption are similar, while all the first iter in each epoch is slow, as the program need to load the data from the very beginning.

Enable or disable cudnn may influence the performance, but I think the difference is tolerable.

**Comment by LQchen1** [comment]
@MzeroMiko  
![image](https://github.com/MzeroMiko/VMamba/assets/93528700/54261cff-243a-4ee9-b6b6-bc3c901b7267)
The configuration file I used is vmambav2_tiny_224.yaml. I contrasted the current log with the author's log and found that the accuracy of EMA was much lower than expected. My EMA has an accuracy of 0.29%, but the author has an accuracy of 6.08% for the same epoch, my emaacc updates very slowly, what is the cause of this, I don't know much about how EMA works.

**Comment by MzeroMiko** [comment]
Oh, it is because the batch_size you use is much bigger than mine. EMA means that in every iter, the parameter is updated with the latest version. The batch_size is smaller, the more frequently the ema parameter is updated, and the higher performance it'll get in a certain epoch. 

But I can not predict what'll happen in the last 50 epochs, as the training start to converge. You may get higher performance with this batchsize.

**Comment by LQchen1** [comment]
@MzeroMiko Thanks for your reply, it is now back to normal. The other question I wanted to know was about CrossScan and CrossMerge, how much benefit would be achieved by rewriting them in Triton compared to Pytoch?
![image](https://github.com/MzeroMiko/VMamba/assets/93528700/227b91c0-357d-4ea0-b34c-e06430243ac3)

**Comment by LQchen1** [comment]
@MzeroMiko Thanks for your reply, it is now back to normal. 
![image](https://github.com/MzeroMiko/VMamba/assets/93528700/6ff118c4-bb8e-486e-87bf-c65edcc53634)
